mongo_connector/doc_managers/algolia_doc_manager.py

```diff
@@ -250,7 +250,8 @@ def apply_update(self, doc, update_spec):
                     doc[attr] = None
         return doc
 
-    def update(self, doc, update_spec, namespace = None, timestamp = None):
+    def update(self, document_id, update_spec, namespace = None, timestamp = None):
+        doc = self.index.getObject(str(document_id))
```

Comment:
It's a shame Mongo doesn't provide the document here, fetching it from the index is very inefficient :/ From what I see in the ES/Solr connector it's unfortunately the way to go :/